<!-- -*- mode: XML -*- -->

<launch>
   <!-- This launch roscore -->
   <master auto="start"/>

  <!-- teresa driver -->  

   <node pkg="teresa_driver" type="teresa_node" name="teresa_node" output="screen" respawn="false">
   <param name="board1" type="str" value="/dev/ttyUSB0"/>
   <param name="board2" type="str" value="/dev/ttyUSB1"/>
   <param name="base_frame_id" type="str" value="/base_link"/>	  
   <param name="odom_frame_id" type="str" value="/odom"/>  
   <param name="head_frame_id" type="str" value="/teresa_head"/>
   <param name="stalk_frame_id" type="str" value="/stalk_head"/>                  
   <param name="freq" type="double" value="50.0" />               
   <param name="using_imu" type="int" value="0"/>                   
   <param name="simulation" type="int" value="1"/>
   <param name="publish_temperature" type="int" value="1"/>
   <param name="publish_buttons" type="int" value="1"/>
   <param name="publish_volume" type="int" value="1"/>
   <param name="publish_power_diagnosis" type="int" value="1"/>
   <param name="number_of_leds" type="int" value="38"/>
   <param name="initial_dcdc_mask" type="int" value="255"/>
   </node>

   <!-- joy node -->
   <node respawn="true" pkg="joy" type="joy_node" name="joy_node">
   <param name="dev" value="/dev/input/js1"/>
   </node>
  
    <!-- teresa joystick driver -->
   <node pkg="teresa_driver" type="teresa_teleop_joy" name="teleop" output="screen" respawn="false">
   </node>

<!-- Rviz -->
  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find teresa_driver)/rviz/teresa_sim.rviz"/>

</launch>
